canopen/objectdictionary/eds.py

```diff
@@ -86,8 +86,10 @@ def import_eds(source, node_id):
 
     if eds.has_section("DeviceComissioning"):
         od.bitrate = int(eds.get("DeviceComissioning", "Baudrate")) * 1000
-        od.node_id = int(eds.get("DeviceComissioning", "NodeID"), 0)
-        node_id = node_id or od.node_id
+
+        if node_id is None:
+            node_id = int(eds.get("DeviceComissioning", "NodeID"), base=0)
```

Comment:
I wonder what happens when the option is not set within the section.  https://docs.python.org/3/library/configparser.html#configparser.ConfigParser.get is unclear on whether an exception is raised.

Reply:
A `DeviceComissioning` section without a `NodeID` option generates a `NoOptionError` exception.

Reply:
It's also unclear in the CiA306 spec whether the options are required, but I'd assume they are optional.  Thus better to explicitly ignore `KeyError` or whatever exception occurs here.  Of course, a `fallback=None`  argument with conditional assignment to the OD would work without caring about exceptions.

Reply:
Sure, but `int(None)` gives you a `TypeError`.

Reply:
That's what I meant with "conditional assignment".  If the option is not present, don't try converting at all.  Whether you detect that by catching the exception, or the `None` fallback value, is up to you.

Reply:
Addressed in ac79a35c792fc60bb0fabad56e60f4d659747a1b.